### PR TITLE
added priority annotation to xml export

### PR DESCRIPTION
**Pull request by mcodescu**
Added a priority attribute to annotated sentences, which is then exported to XML. We need this fast for Coinvent.

### CODE REVIEW

File: Common/AS_Annotation.der.hs [hunk 1]
```diff
@@ -265,9 +267,17 @@ identAnno str an = case an of
 hasIdentAnno :: String -> Annoted a -> Bool
 hasIdentAnno str a = any (identAnno str) $ l_annos a ++ r_annos a
 
+getPriority :: [Annotation] -> Maybe String
+getPriority annos = 
+  foldl (\mId anno -> case anno of 
+                       Unparsed_anno (Annote_word "priority") (Group_anno (x:_)) _ -> Just x
```
Comment (cmaeder) [inline Common/AS_Annotation.der.hs]: These cases should just be indented by 4 (or 8) spaces. also:

```
$ checkSource.sh Common/AS_Annotation.der.hs 
Common/AS_Annotation.der.hs:271:20: trailing (1) white space
Common/AS_Annotation.der.hs:272:35: trailing (1) white space
Common/AS_Annotation.der.hs:273:1: too long line (92 chars)
Common/AS_Annotation.der.hs:272:11: put blank after \
Common/AS_Annotation.der.hs:273:77: put blank before :
Common/AS_Annotation.der.hs:273:78: put blank after :
Common/AS_Annotation.der.hs:271:1: Error: Eta reduce
Found:
  getPriority annos
    = foldl
        (\ mId anno ->
           case anno of
               Unparsed_anno (Annote_word "priority") (Group_anno (x : _))
                 _ -> Just x
               _ -> mId)
        Nothing
        annos
Why not:
  getPriority
    = foldl
        (\ mId anno ->
           case anno of
               Unparsed_anno (Annote_word "priority") (Group_anno (x : _))
                 _ -> Just x
               _ -> mId)
        Nothing
```

**Comment by cmaeder** [comment]
also Static.ToXml has trailing white space.

**Comment by cmaeder** [comment]
You my also want to add the priorities to Static.toJson. (Please call a pull after my push.)

**Comment by mcodescu** [comment]
Done, but just by guessing. Feel free to correct if necessary.

**Comment by cmaeder** [comment]
These changes in Common are not really necessary. The construction can directly be used in Static.
Did you test an example? How should a priority annotation look like? Otherwise the change looks fine to me. :+1:

**Comment by mcodescu** [comment]
here is an example of output: 
```
<Axioms>
      <Axiom name="fixpoint_c" range="/home/mcodescu/repos/HETS/testXML.dol:3.9-5.11" priority="1">
        <Text>. f(c) = c</Text>
        <Symbol kind="op" name="c" iri="c" range="/home/mcodescu/repos/HETS/testXML.dol:5.6">op c : s</Symbol>
        <Symbol kind="op" name="f" iri="f" range="/home/mcodescu/repos/HETS/testXML.dol:5.4">op f : s -&gt; s</Symbol>
        <Symbol kind="sort" name="s" iri="s" range="/home/mcodescu/repos/HETS/testXML.dol:4.15">sort s</Symbol>
        <AST>
          <Equation>
            <Application>
              <Qual_op_name>
                <id>f</id>
                <Op_type>
                  <Total />
                  <list>
                    <id>s</id>
                  </list>
                  <id>s</id>
                  <range />
                </Op_type>
                <range>/home/mcodescu/repos/HETS/testXML.dol:5.7</range>
              </Qual_op_name>
              <list>
                <Application>
                  <Qual_op_name>
                    <id>c</id>
                    <Op_type>
                      <Total />
                      <list />
                      <id>s</id>
                      <range />
                    </Op_type>
                    <range>/home/mcodescu/repos/HETS/testXML.dol:5.6</range>
                  </Qual_op_name>
                  <list />
                  <range>/home/mcodescu/repos/HETS/testXML.dol:5.6</range>
                </Application>
              </list>
              <range>/home/mcodescu/repos/HETS/testXML.dol:5.7</range>
            </Application>
            <Strong />
            <Application>
              <Qual_op_name>
                <id>c</id>
                <Op_type>
                  <Total />
                  <list />
                  <id>s</id>
                  <range />
                </Op_type>
                <range>/home/mcodescu/repos/HETS/testXML.dol:5.11</range>
              </Qual_op_name>
              <list />
              <range>/home/mcodescu/repos/HETS/testXML.dol:5.11</range>
            </Application>
            <range>/home/mcodescu/repos/HETS/testXML.dol:5.9</range>
          </Equation>
        </AST>
      </Axiom>
```

**Comment by cmaeder** [comment]
How does the line look like in the .dol file?

**Comment by mcodescu** [comment]
In case you missed my edit of the previous comment

```
spec S =
  sort s 
  op c: s
  op f : s -> s
  .f(c) = c %(fixpoint_c)% %priority(1)%
```

**Pull request merged**